subcommands/keys/rotate_root.go

```diff
@@ -314,3 +324,45 @@ func getOfflineCreds(credsFile string) OfflineCreds {
 	}
 	return files
 }
+
+func syncProdRoot(factory string, root client.AtsTufRoot, creds OfflineCreds) error {
+	fmt.Println("= Populating production root version")
+
+	// Bump the threshold
+	root.Signed.Roles["targets"].Threshold = 2
+
+	// Sign with the same keys used for the ci copy
+	var signers []TufSigner
+	for _, sig := range root.Signatures {
+		key, ok := root.Signed.Keys[sig.KeyID]
+		if !ok {
+			// Root key was rotated, this pub key is previous version
+			prev, err := api.TufRootGetVer(factory, root.Signed.Version-1)
+			if err != nil {
+				return err
+			}
+			key = prev.Signed.Keys[sig.KeyID]
+		}
+		pkey, err := findPrivKey(key.KeyValue.Public, creds)
+		if err != nil {
+			return err
+		}
+		signers = append(signers, TufSigner{
+			Id:  sig.KeyID,
+			Key: pkey,
+		})
+	}
+	if err := signRoot(&root, signers...); err != nil {
+		return err
+	}
+
+	bytes, err := json.Marshal(root)
+	if err != nil {
+		return err
+	}
+	_, err = api.TufProdRootPost(factory, bytes)
```

Comment:
When checking a validation logic on API side I've found that we expect prod_root_version <= ci_root_version... but shouldn't we expect them to be equal ?
https://git.foundries.io/development/cloudplatforms/infrastructure/ota-lite/-/blob/master/ota_lite/api/reposerver.py#L262

I'm sorry that I didn't find it during an ota-lite code review.

Also, in case if a prod root role with the same version already exists - I'd expect a 409 status code (currently it returns 400 as I understand).

Reply:
> we expect prod_root_version <= ci_root_version... but shouldn't we expect them to be equal ?

good catch. that's left over logic from the original backfill approach.

> we expect prod_root_version <= ci_root_version... but shouldn't we expect them to be equal ?

sounds reasonable.

Reply:
https://git.foundries.io/development/cloudplatforms/infrastructure/ota-lite/-/merge_requests/109